src/transactions/deso-tokens.ts

```diff
@@ -41,10 +49,50 @@ export type ConstructBurnDeSoTokenRequestParams =
       | 'CoinsToBurnNanos'
     >
   >;
-export const burnDeSoToken = (
+export const burnDeSoToken = async (
   params: ConstructBurnDeSoTokenRequestParams,
-  options?: RequestOptions
+  options?: TxRequestOptions
 ): Promise<ConstructedAndSubmittedTx<DAOCoinResponse>> => {
+  if (options?.checkPermissions !== false) {
+    const txWithFee = getTxWithFeeNanos(
+      params.UpdaterPublicKeyBase58Check,
+      new TransactionMetadataDAOCoin(),
+      {
+        // TODO: I'm not sure exactly what outputs are needed here... for the time
+        // being I'm just adding a static 1500 nanos to make sure the derived key
+        // transaction can be submitted.
+        // Outputs: ...,
+        ExtraData: params.ExtraData,
+        MinFeeRateNanosPerKB: params.MinFeeRateNanosPerKB,
+        TransactionFees: params.TransactionFees,
+      }
+    );
+
+    if (!isMaybeDeSoPublicKey(params.ProfilePublicKeyBase58CheckOrUsername)) {
+      return Promise.reject(
+        'must provide profile public key, not username for ProfilePublicKeyBase58CheckOrUsername when checking dao coin transfer permissions'
+      );
+    }
+
+    const txnLimitCount =
+      options?.txLimitCount ??
+      identity.transactionSpendingLimitOptions.TransactionCountLimitMap
+        ?.DAO_COIN ??
+      1;
```

Comment:
@stas-kh - can we update this to be similar to lines 318-323 please? It will be more correct this way.